src/eredis_sync.erl

```diff
@@ -15,32 +15,35 @@
 -type response() :: {ok, iodata()} | {error, error()}.
 -type response_list() :: [response()].
 
--spec connect(inet:ip_address(), inet:port_number()) -> {ok , conn()} | {error, error()}.
+-spec connect(inet:ip_address() | iodata(), inet:port_number()) -> {ok , eredis_sync:conn()} | {error, error()}.
 
 connect(Host, Port) ->
   connect(Host, Port, ?DEFAULT_TIMEOUT).
 
--spec connect(inet:ip_address(), inet:port_number(), timeout()) -> {ok , conn()} | {error, error()}.
+-spec connect(inet:ip_address() | iodata(), inet:port_number(), timeout()) -> {ok , eredis_sync:conn()} | {error, error()}.
```

Comment:
Why to specify `iodata()` when the function actually receives only binaries?

Reply:
What about host value like `"127.0.0.1"`? `inet:ip_address()` and `binary()` don't cover this case.

Reply:
We should probably take signature from https://erlang.org/doc/man/gen_tcp.html#connect-3

Reply:
I think it great idea. Now I use same specifications as gen_tcp:connect and dialyzer is passing successfully.